app/components/Form/index.js

```diff
@@ -0,0 +1,271 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Broadcast from 'utils/Broadcast';
+
+/**
+ * Form component
+ *
+ * It handles all the form state, and provides information and actions to
+ * the children components using context.
+ *
+ * It takes the following props:
+ * - fields (required): Array of field names.
+ * - initialValues:     Map of initial values.
+ * - onSubmit:          Function to call on successful form submit.
+ *                      If not provided, the submit logic can be handled in the
+ *                      component.
+ * - onFormDataChange:  Function to call with the form data when it changes.
+ * - validate:          Function to validate the form on every change, if not
+ *                      provided the form is always valid.
+ *
+ * The following props will be passed in a context Broadcast:
+ * - fields:            Object with the value, blurred state, and errors for every field.
+ *                      It also has function to attach the fields to the form state.
+ * - error:             A generic error for the entire form (from the `_error` field returned
+ *                      in the validation function).
+ * - valid:             True is the form is valid.
+ * - handleSubmit:      When called, the form will be validated and the provided 'onSubmit' function
+ *                      will be called.
+ * - initializeForm:    Initialize the form with the provided field values.
+ *
+ * To attach the form state to your form fields, you should use a component which takes form data
+ * from the context Broadcast. (check `components/Field`)
+ *
+ */
+class Form extends React.Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    fields: PropTypes.array.isRequired,
+    initialValues: PropTypes.object,
+    onSubmit: PropTypes.func,
+    onFormDataChange: PropTypes.func,
+    validate: PropTypes.func,
+  };
+
+  static defaultProps = {
+    children: null,
+    initialValues: {},
+    onSubmit: null,
+    onFormDataChange: null,
+    validate: null,
+  };
+
+  static childContextTypes = {
```

Comment:
I feel this should also be in a hoc to make it maintainable.

Reply:
Done. 

Also I updated the `<Form>` HOC. Now `Broadcast`, `provideContextBroadcast` and `consumeContextBroadcast` are all that's needed to abstract sending data through context.